libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_disk.py

```
@@ -157,12 +158,126 @@ def _check_disk(target):
+        The selected device must not back the host root filesystem.
+        """
+        root_source = process.run(
+            "findmnt -n -o SOURCE /",
+            shell=True,
+            ignore_status=False
+        ).stdout_text.strip()
+        if not root_source:
+            test.cancel("Unable to determine host root source device")
+
+        root_source = root_source.split("[", 1)[0].strip()
+        root_parent_result = process.run(
```

Ruff (0.15.20)

[error] Function call with `shell=True` parameter identified, security issue

(S604)